Framework/API/src/ScriptBuilder.cpp

```diff
@@ -87,13 +88,23 @@ void ScriptBuilder::writeHistoryToStream(
     }
   } else {
     // create the string for this algorithm
-    os << buildAlgorithmString(*algHistory);
-    if (m_timestampCommands) {
-      os << " # " << algHistory->executionDate().toISO8601String();
+    if (!m_ignoreGroups) {
+      createStringForAlg(os, algHistory);
+    } else if (m_ignoreGroups && algHistory->name() != "GroupWorkspaces") {
+      createStringForAlg(os, algHistory);
```

Comment:
Could this `if`-`else` construct be simplified to:
```c++
if (!m_ignoreGroups || algHistory->name() != "GroupWorkspaces") {
  createStringForAlg(os, algHistory);
}
```
?

Reply:
It can, good spot!